src/chapters/strings/encoding-characters.rst

```diff
@@ -0,0 +1,215 @@
+Encoding Characters
+===================
+
+.. index:: ! bit, character encoding
+
+If you had microscope powerful enough to view the data stored on a computer's
+hard drive, or in its memory, you would see lots of 0s and 1s. Each such 0 and
+1 is known as a **bit**. A bit is a unit of measurement, like a meter or a
+pound. Collections of computer data are measured in bits; every letter, image,
+and pixel you interact with on a computer is represented by bits.
+
+We work with more complex data when we program, including numbers and strings.
+This section examines how such data is represented within a computer.
+
+Representing Numbers
+--------------------
+
+.. index:: ! byte
+
+.. index::
+   pair: number; binary
+   pair: number; decimal
+
+A **byte** is a set of 8 bits. Bytes look like 00101101 or 11110011, and they
+represent a **binary number**, or a base-2 number. A binary number is a number
+representation that uses only 0s and 1s. The numbers that you are used to,
+which are built out of the integers 0...9, are **decimal numbers**, or base-10
+numbers.
+
+Since each bit can have one of two values, each byte can have one of
+2\ :sup:`8` = 256 different values.
+
+It may not be obvious, but every decimal integer can be represented as a binary
+integer, and vice versa. There are 256 different values a byte may take, each
+of which can be used to represent a decimal integer, from 0 to 255.
+
+.. admonition:: Note
+
+   We will not go into binary to decimal number conversion. If you are
+   interested in learning more, there are `many <https://www.csetutor.com/how-to-convert-binary-to-decimal-examples/>`_ `tutorials <https://www.youtube.com/watch?v=wPvI19DmWQw>`_ `online <https://www.khanacademy.org/math/algebra-home/alg-intro-to-algebra/algebra-alternate-number-bases/v/decimal-to-binary>`_ that can show you the way.
```

Comment:
this is a cute and funny way to present these resources! the first link (on "many" ) doesn't work though - i get a connection refused. maybe a typo?